token-client/src/test/java/com/sap/cloud/security/client/DefaultTokenClientConfigurationTest.java

```
@@ -0,0 +1,121 @@
+
+  private DefaultTokenClientConfiguration config;
+
+  @BeforeEach
+  public void setUp() {
+    config = new DefaultTokenClientConfiguration();
+  }
+
+  @Test
+  public void testStaticGetterAndSetter() {
+    DefaultTokenClientConfiguration.setConfig(config);
+    assertThat(DefaultTokenClientConfiguration.getConfig()).isEqualTo(config);
+  }
+
+  @Test
+  public void defaultValues_areSetCorrectly() {
+    assertThat(config.isRetryEnabled()).isFalse();
+    assertThat(config.getMaxRetryAttempts()).isEqualTo(3);
+    assertThat(config.getRetryDelayTime()).isEqualTo(1000L);
+    assertThat(config.getRetryStatusCodes())
+        .containsExactlyInAnyOrder(408, 429, 500, 502, 503, 504);
+  }
+
+  @Test
+  public void setRetryEnabled_updatesValue() {
+    config.setRetryEnabled(true);
+    assertThat(config.isRetryEnabled()).isTrue();
+  }
+
+  @Test
+  public void setMaxRetryAttempts_updatesValue() {
+    config.setMaxRetryAttempts(5);
+    assertThat(config.getMaxRetryAttempts()).isEqualTo(5);
+  }
+
+  @Test
+  public void setRetryDelayTime_updatesValue() {
+    config.setRetryDelayTime(2000L);
+    assertThat(config.getRetryDelayTime()).isEqualTo(2000L);
+  }
+
+  @Test
+  public void setRetryStatusCodes_withString_updatesValue() {
+    config.setRetryStatusCodes("400,401");
+    assertThat(config.getRetryStatusCodes()).containsExactlyInAnyOrder(400, 401);
+  }
+
+  @Test
+  public void setRetryStatusCodes_withIntegerSet_updatesValue() {
+    config.setRetryStatusCodes(Set.of(400, 401));
+    assertThat(config.getRetryStatusCodes()).containsExactlyInAnyOrder(400, 401);
+  }
+
+  @Test
+  public void setRetryStatusCodes_withEmptyString() {
+    config.setRetryStatusCodes("");
+    assertThat(config.getRetryStatusCodes()).isEmpty();
+  }
+
+  @Test
+  public void setRetryStatusCodes_withInvalidString() {
+    config.setRetryStatusCodes("abc,123");
+    assertThat(config.getRetryStatusCodes()).containsExactlyInAnyOrder(123);
```

not sure if we should really silently ignore malformed entries. did you think about failing in this case?